File: csharp/exercises/Webshop.Service/NotificationService.cs
namespace Webshop.Service;

public interface INotificationService
{
    void SendMail(string recipientEmail, string message);
}

public class NotificationService : INotificationService
{
    public void SendMail(string recipientEmail, string message)
    {
        // Send mail to the specified email-address
        // Uses an email-server which is not available from our developer environment
        Thread.Sleep(int.MaxValue);
    }
}

File: java/exercises/src/main/java/Webshop/Service/Common/NotificationService.java
package java.Webshop.Service.Common;

interface INotificationService {
    void sendMail(String recipientEmail, String message);
}

public class NotificationService implements INotificationService {
    public void sendMail(String recipientEmail, String message) {
        // Simulate sending mail (since the email server is not available)
        try {
            Thread.sleep(Integer.MAX_VALUE);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }
}